pylint/checkers/imports.py

```diff
@@ -914,11 +914,27 @@ def _add_imported_module(self, node: ImportNode, importedmodname: str) -> None:
 
     def _check_preferred_module(self, node: ImportNode, mod_path: str) -> None:
         """Check if the module has a preferred replacement."""
-        if mod_path in self.preferred_modules:
+
+        mod_compare = [mod_path]
+        # build a comparison list of possible names using importfrom
+        if isinstance(node, astroid.nodes.node_classes.ImportFrom):
```

Comment:
This should always be true based on the typing. Is the typing of the signature incorrect or can we remove this check?

Reply:
Astroid provides an [`ImportFrom`](https://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.ImportFrom) as well as an [`Import`](https://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.Import) node. This if case was added in for `from ... import ...` cases where a comparison string was necessary to match from the preferred-modules pairs involving submodules with "dot" notation (for ex. `"module.submodule"` str construction). Otherwise with `Import` cases we only need the module name.

Reply:
Oh my, I misread `ImportNode` for `ImportFrom`. I'm stupid.. Thanks for this!

Reply:
No worries at all! Thank you for calling it forward just the same.